### DISCUSSION TITLE
Merge two yaml file selecting keys using wildcard character and removing duplicates

### DISCUSSION THREAD

**Discussion by Galileo1** [mikefarah/yq#1742] [General]
I want to merge two yaml files using yq in bash script.Selecting only specific keys values , removing duplicates and building a string.

file1.yaml
```
FT1: true
FT2: true
```

file2.yaml

```
FT1: false
FT2: false
FT3: false
FT4: false
Blah1: abc
Blah2: efg
```

Desired Output:

Trying to build a command like below.

- Filter any keys that starts with (FT*)
- join the key=values
- removing FT1, FT2 from file2.yaml
- And building a command for each key=value pair appending it with --from-literal=

`--from-literal=FT1=true --from-literal=FT2=true --from-literal=FT3=false --from-literal=FT4=false`

What I have tried so far:
`yq eval '. | to_entries| .[] | select(.key == "FT*") | [.key, .value] | join("=") file1.yaml`

The above command output is as below. Not sure how to merge them , remove duplicates from file2 and build the command.

file1.yaml
`
FT1=true
FT2=true
`

**Comment by mikefarah** [comment]
TLDR:

`yq '. * load("data1.yaml") | to_entries | [.[] | select(.key == "FT*") | "--from-entries=" + .key + "=" + .value] | join(" ")' data2.yaml `

Explanation:
- `. * load("data1.yaml")` merge data1 on top of data2, overwriting any duplicate keys (switch the order if you want it the other way around)
- Convert the map `to_entries` like you had
- `.[] | select(.key == "FT*")` select all the entries with FT
- `"--from-entries=" + .key + "=" + .value` construct the desired string for each entry
- `[.[] .... .value]` wrap the whole entries expression in an array - so you can join it with spaces :)

Edit: formatting
